packages/storage/src/providers/s3/apis/internal/list.ts

```diff
@@ -46,17 +45,8 @@ interface ListInputArgs {
 
 export const list = async (
 	amplify: AmplifyClassV6,
-	input:
-		| ListAllInput
-		| ListPaginateInput
-		| ListAllWithPathInput
-		| ListPaginateWithPathInput,
-): Promise<
-	| ListAllOutput
-	| ListPaginateOutput
-	| ListAllWithPathOutput
-	| ListPaginateWithPathOutput
-> => {
+	input: ListInput | ListInternalInput,
```

Comment:
I don't think we need to touch the deprecated Gen1 interface. Should it be something like this?
```suggestion
	input: ListAllInput | ListPaginatedInput | ListAllWithPathInternalInput | ListPaginateWithPathInternalInput,
```

or 
```suggestion
  input: ListAllInput | ListPaginatedInput | ListInternalInput
```